Fixing incorrect normal map when using triplanar world mapping and mesh rotation

RPicster opened the pull request:
Fixes #82311

dalexeev committed:
GDScript: Fix incorrect error message for utility functions

mandryskowski committed:
Fresnel should darken the SSR reflections instead of blending them with specular light.

Commit:
Fix OGG audio loop offset pop.

Co-authored-by: MJacred <loesch.benny92@gmx.de>
Co-authored-by: Ellen Poe <ellen.h.poe@gmail.com>
Co-authored-by: Michael Wörner <mwoerner@semw-software.com>

raulsntos committed:
C#: Improve diagnostic messages and add help link

- Reword diagnostic messages.
- Add help link to diagnostics.

QbieShay committed:
Particle internal refactor and additions for more artistic control

Co-authored-by: Hugo Locurcio <hugo.locurcio@hugo.pro>
Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>
Co-authored-by: Raul Santos <raulsntos@gmail.com>
Co-authored-by: Mew Pur Pur <85438892+MewPurPur@users.noreply.github.com>
Co-authored-by: Clay John <claynjohn@gmail.com>

bruvzg committed:
[Bitmap fonts] Add support for scaling.

MewPurPur committed:
Improve VisibleOnScreen classes docs

georgwacker committed:
Add project settings for AVAudioSessionCategory on iOS

Rindbee committed:
Defer updating the animations Tree in SpriteFramesEditor to avoid crashes

Previously, clicking the LMB while renaming an animation could cause
`SpriteFramesEditor::_update_library(false)` to be called during
`Tree::propagate_mouse_event()`. This may cause a crash.

We can defer updates to the editor interface to avoid calling
`Tree::create_item()` at the wrong time.

Enables `SpriteFramesEditor::_select_animation()` to be able to undo/redo

ajreckof committed:
Fix ColorPicker deferred mode not working for sliders.

ErezShahaf committed:
Fix `Input.is_action_just_pressed` flicker on joypad axes

Pressed tick assignments were in the wrong scope, resulting in updating
`pressed_frame` even when it shouldn't and therefore the `just_pressed`
would return true every time that the strength changes and not only when
there's a new valid press.

Fixes #81975.

RPicster committed:
Fixing incorrect normal map when using triplanar world mapping and mesh rotation

dervus committed:
Fix grid snapping for box shape gizmos

"Extruding" box face (i.e. moving it without holding Alt key) will now
always snap new box size exactly to the nearest snap step.

Faless committed:
[MP] Optimize multiplayer NodePath caching

Only use paths during network transfer.
Use ObjectID instead of NodePaths for storing the Node <-> NetID
relations locally.

Faless committed:
[MP] Optimize internal authority checks

We already know which MultiplayerAPI a certain Node uses, so we don't
need to retrieve it via SceneTree every time.

Faless committed:
[MP] Avoid unnecessary internal ref/unrefs

Access the various internal components (cache/replicator) via pointer,
to avoid unnecessary overhead.

KoBeWi committed:
Don't remove favorite files in EditorFileDialog

akien-mga committed:
openxr: Sync with upstream 1.0.30

paulloz committed:
Error handling for `FileAccess.get_file_as_*`

- Assign last error in said `FileAccess.get_file_as_bytes` and `FileAccess.get_file_as_string`
- Document error handling for said methods

Faless committed:
[MP] Optimize internal SceneReplicationConfig access

Use direct pointer addressing to avoid unnecessary refs/unrefs

raulsntos committed:
C#: Add checks to Android export

- Add .NET 7.0 TFM when the platform is Anroid to the created csproj.
- Prevent exporting to Android when the architecture is not supported.

bruvzg committed:
Re-add optional OIDN denoise as an external executable.

clayjohn commented on scene/resources/material.cpp:
```diff
@@ -1062,12 +1062,12 @@ void BaseMaterial3D::_update_shader() {
 		code += "	TANGENT+= vec3(1.0,0.0,0.0) * abs(NORMAL.y);\n";
 		code += "	TANGENT+= vec3(1.0,0.0,0.0) * abs(NORMAL.z);\n";
-		code += "	TANGENT = normalize(TANGENT);\n";
+		code += "	TANGENT = normalize(TANGENT) * MODEL_NORMAL_MATRIX;\n";
```
The extra code should only be added when using world triplanar on UV1 (world triplanar on UV2 doesn't matter because UV2 is never used for normal mapping)

RPicster replied:
alright, I'll add this conditional as soon as I find the opportunity :)

clayjohn reviewed (commented):
This looks very fishy to me. What this does, mathematically speaking, is multiply the world space tangent and by the transposed model matrix. Which shouldn't really be a meaningful transformation. 

Testing locally I can confirm it works, but the intuition is escaping me. 

In 3.x we used ``world_vertex_coords`` with this exact code block in order to generate tangent and bitangent in world space. ``world_vertex_coords`` then translates them into view space automatically. 

In master, we are creating tangent and bitangent in local space, then Godot automatically transforms them from local space into view space. We get issues when local space and world space no longer line up.

Here we create a tangent and bitangent in local space, then transform them into something resembling world space. And somehow it works. 

I can't escape the conclusion that the correct code should be:
```
vec3 normal = MODEL_NORMAL_MATRIX * NORMAL;
TANGENT = vec3(0.0,0.0,-1.0) * abs(normal.x);
TANGENT+= vec3(1.0,0.0,0.0) * abs(normal.y);
TANGENT+= vec3(1.0,0.0,0.0) * abs(normal.z);
TANGENT = inverse(MODEL_NORMAL_MATRIX)* normalize(TANGENT);
BINORMAL = vec3(0.0,1.0,0.0) * abs(normal.x);
BINORMAL+= vec3(0.0,0.0,-1.0) * abs(normal.y);
BINORMAL+= vec3(0.0,1.0,0.0) * abs(normal.z);
BINORMAL = inverse(MODEL_NORMAL_MATRIX)* normalize(BINORMAL);
```
(This code works as well)

_Edit: I think I figured it out. Since in the MRP the MODEL_NORMAL_MATRIX is orthogonal, its transpose and inverse are equal. By multiplying it on the right hand side, you are doing the same as a multiplication between the vector and the transpose of the matrix. Which means that you can view it as multiplying by the INV_MODEL_NORMAL_MATRIX and transforming from world space to model space. This appears to work because the normal is always the same for a planemesh, it doesn't work for more complex objects. Relying on this right-hand multiplication also fails for non-orthogonal matrices (i.e. non-uniform scale). So unfortunately, we are stuck with the inverse, or with doing the calculation in world space._

dsnopek committed:
GDExtension: Use correct return pointer for validated calls that return `Variant`

SaracenOne committed:
Set new RestFixer tracks as imported

bruvzg committed:
[iOS] Fix build on Xcode 14 and older.

akien-mga committed:
Merge pull request #82762 from raulsntos/dotnet/android-ux

C#: Add checks to Android export

akien-mga committed:
Merge pull request #82893 from m4gr3d/godot_android_lib_improvements

Cleanups and improvements to the Godot Android library api

akien-mga committed:
Merge pull request #83062 from bruvzg/no_dup_es

Do not apply extra spacing twice.

akien-mga committed:
Merge pull request #83076 from SaracenOne/make_rest_fixer_tracks_imported

Set new SkeletonRestFixer tracks as imported

akien-mga committed:
Merge pull request #83088 from bruvzg/ios_pre_xc15

[iOS] Fix build on Xcode 14 and older.

dalexeev committed:
Editor: Fix some issues with `EditorHelpTooltip`

groud committed:
Fixes undo/redo in tileset polygon editor

akien-mga committed:
Cleanup various repository documentation files

- Reformat logo license as a plain text file.
- Fix outdated links or references to SFC or Visual Script.
- Tweak contents of `CONTRIBUTING.md` to highlight contributor docs more
  prominently, and make it easier to parse.
- Tweak formatting and contents in `thirdparty/README.md` for consistency.

bruvzg committed:
[macOS] Use occlusionState instead of isOnActiveSpace to determine when window is drawable.

akien-mga committed:
Merge pull request #82973 from dsnopek/gdextension-linux-favor-local-symbols

On Linux, favor local symbols when loading a shared library

akien-mga committed:
Merge pull request #83049 from dalexeev/gds-docgen-fix-return-metatype-regression

GDScript DocGen: Fix regression with return metatypes

akien-mga committed:
Merge pull request #83050 from k0T0z/fix-unnecessary-include

Deleting unnecessary include in GDScriptParser

akien-mga committed:
Merge pull request #83065 from KoBeWi/tree_works_in_mysterious_ways

Allow clicking buttons of non-selectable TreeItems

akien-mga committed:
Merge pull request #83093 from groud/fix_polygon_editor_undo_redo

Fixes undo/redo in tileset polygon editor

kleonc committed:
Fix incorrect vertex data size calculation in ImmediateMesh

joao-pedro-braz committed:
Fix highlight in Search panel being offseted

This commit fixes the match highlight in the Find In Files
result panel being slightly offseted.

akien-mga committed:
Merge pull request #80817 from garychia/reparent_node

Avoid creating any useless undo action when dragging nodes in place

akien-mga committed:
Merge pull request #82777 from Faless/mp/perfs

[MP] Various performance optimizations

akien-mga committed:
Merge pull request #83095 from akien-mga/cleanup-repo-docs

Cleanup various repository documentation files

dsnopek committed:
GDExtension: Fix incorrect error message about vararg methods

Repiteo committed:
Fix dotnet dev_mode regression

akien-mga committed:
Merge pull request #79527 from QbieShay/qbe/particles-rework

Particle internal refactor and additions for more artistic control

akien-mga committed:
Merge pull request #81034 from garychia/open_folder

FileDialog: Avoid selecting the first item automatically in Open Folder Mode

akien-mga committed:
Merge pull request #83054 from dsnopek/gdextension-validated-call-always-ret-pointer

GDExtension: Use correct return pointer for validated calls that return `Variant`

akien-mga committed:
Merge pull request #83094 from dalexeev/editor-fix-some-editor-help-tooltip-issues

Editor: Fix some issues with `EditorHelpTooltip`

akien-mga committed:
Merge pull request #83096 from bruvzg/macos_leak

[macOS] Use occlusionState instead of isOnActiveSpace to determine when window is drawable.

akien-mga committed:
Merge pull request #83100 from kleonc/immediate-mesh-vertex-buffer-size-fiz

Fix incorrect vertex data size calculation in `ImmediateMesh`

akien-mga committed:
Merge pull request #83101 from joao-pedro-braz/fix-find-in-files-selection-offset

Fix highlight rect in "Whole search" being slightly offseted

akien-mga committed:
Merge pull request #83107 from dsnopek/validated-method-vararg-message

GDExtension: Fix incorrect error message about vararg methods

akien-mga committed:
Merge pull request #83114 from Repiteo/fix-dotnet-dev-build-regression

C#: Fix MSVC dotnet `dev_mode` regression

astillich committed:
Fix editing exports in a base script not propagating the change to the opened property editor

This patch fixes the user having to navigate away from the selected node which has the derived script attached and back to see the changes of the base script exports reflected in the property editor.

akien-mga committed:
Merge pull request #79169 from reduz/debug-canvas-item-redraw

Debug CanvasItem redraw

radiantgurl committed:
Fix `RefCounted.unreference()` documentation providing wrong info.

jsjtxietian committed:
Close shader in Shader Editor tab when deleting shader file in FileSystem panel

bitsawer committed:
Fix BaseMaterial3D update with certain material settings

BastiaanOlij committed:
Fix compatibility shadow size not being initialised

KoBeWi committed:
Disable update spinner when debug redraw is active

groud committed:
Warn users when TileMap is set as Y-sorted but no layer is

akien-mga committed:
CI: Switch mesa PPA from kisak-mesa to turtle

May solve a weird name clash we seem to have on CI since yesterday.

dalexeev committed:
GDScript: Add autocompletion for static variables accessed via class

kleonc committed:
Update TileMap layer draw index when it's dirty

pancelor committed:
Fix tilemap live editing while game is running

akien-mga committed:
Merge pull request #83141 from BastiaanOlij/fix_shadow_size

Fix compatibility shadow size not being initialised

akien-mga committed:
Merge pull request #83143 from KoBeWi/you_spin_me_round_not

Disable update spinner when debug redraw is active

akien-mga committed:
Merge pull request #83144 from groud/tilemap_y_sort_warning

Warn users when TileMap is set as Y-sorted but no layer is

akien-mga committed:
Merge pull request #83145 from bitsawer/fix_basematerial3d_update

Fix BaseMaterial3D update with certain material settings

akien-mga committed:
Merge pull request #83147 from akien-mga/ci-ppa-kisak-turtle

CI: Switch mesa PPA from kisak-mesa to turtle

akien-mga committed:
Merge pull request #83151 from kleonc/tile-map-layer-draw-index-update

Update `TileMap` layer draw index when it's dirty

Faless committed:
[MP] Fix synchronizer init and reset

Fix set_multiplayer_authority not resetting the synchronizer.

Fix the reset function not clearing the watchers state.

Skip wrap around check for the first sync packet after reset.

Faless committed:
[Web] Improve emscripten "locateFile" glue.

Be more selective on what we rewrite, since in recent emscripten
versions loading dynamic libraries relies on it.

akien-mga committed:
Merge pull request #79921 from mandryskowski/ssr_transparency

Make Fresnel darken SSR instead of blending with specular

akien-mga committed:
Merge pull request #82557 from RadiantUwU/patch-2

Fix `RefCounted.unreference()` documentation providing wrong info.

akien-mga committed:
Merge pull request #82595 from paulloz/fix-fileaccess-error-handling

Error handling for `FileAccess.get_file_as_*`

akien-mga committed:
Merge pull request #82832 from bruvzg/oidn_external_exe

Re-add optional OIDN denoise as an external executable.

akien-mga committed:
Merge pull request #82986 from OmarShehata/fix-external-editor-hot-reload

Fix external editor hot reload for GDScript

akien-mga committed:
Merge pull request #83013 from RobTheFiveNine/documentation/start_button

Add documentation on which buttons JOY_BUTTON_START corresponds to

akien-mga committed:
Merge pull request #83150 from dalexeev/gds-autocomplete-static-var-on-class

GDScript: Add autocompletion for static variables accessed via class

akien-mga committed:
Merge pull request #83165 from Faless/web/locatefile_fixes

[Web] Improve Emscripten `locateFile` glue.

akien-mga committed:
Merge pull request #82100 from bitsawer/fix_shader_globals_override

Fix ShaderGlobalsOverride property handling

akien-mga committed:
Bump version to 4.2-beta

The feature freeze for 4.2 is now official, from here on we'll focus only
on fixing bugs to stabilize the development branch and release 4.2-stable
within a month.

clayjohn committed:
Fix a few cases where surface format is still 32 bits

akien-mga committed:
Merge pull request #83169 from clayjohn/Mesh-format

Fix a few cases where surface format is still 32 bits

akien-mga committed:
Merge pull request #82431 from bitsawer/fix_gpuparticles_free

Fix errors when freeing GPUParticles

Commit:
Fix the timeframe when the Android gestures properties are retrieved.

Previous logic was retrieving them prior to them being defined in `main.cpp`

timothyqiu committed:
Don't auto translate theme type list

clayjohn committed:
Sanitize tangents when creating mesh surfaces to avoid triggering the compressed mesh path in the shader

bruvzg committed:
Add error messages to the native menu and file dialogs callback.

jsjtxietian committed:
Fix parameter shader node not declared when only connected to a VaryingSetter

Kurble committed:
Put cheaper condition first

Co-authored-by: A Thousand Ships <96648715+AThousandShips@users.noreply.github.com>

jsjtxietian committed:
Fix bool varying's generated code will be modified with flat

Repiteo committed:
Apply missing flag to vector4/projection

alfredbaudisch committed:
Handle "Enter" when the search box is focused

clayjohn committed:
Cleanup instances of using uint32_t for mesh formats

And tidy up some leftovers from the attribute compression PR

dalexeev committed:
Code Editor: Fix regression with using doc comments for code regions

akien-mga committed:
CI: Workaround recently broken add-apt-repository on GHA

Hopefully adding the sources manually still works.

akien-mga committed:
Merge pull request #83214 from akien-mga/ci-linux-mesa-alt-install

CI: Workaround recently broken add-apt-repository on GHA

timothyqiu committed:
Disable port name auto translation in Visual Shader editor

Daylily-Zeleen committed:
Fix missing editor singletons when dump extension api

bruvzg committed:
[macOS] Fix crash when using system default menu shortcuts.

KoBeWi committed:
Swap TileMap and TileSet

Rindbee committed:
Fix saving wrong edited scene state when switching scene tabs

When creating a new scene, the scene tab will actually switch to the newly created one.
Also worth noting that switching scene tabs saves state (by `save_edited_scene_state`).

When trying New Scene again, the previously empty scene tab will be removed.

So the correct logic is: switch the tab save state first, and then remove the previous
empty scene tab. To prevent `current_edited_scene` from being incorrect when saving state.

zaevi committed:
C#: Fix lookup for singleton instance types.

mrjustaguy committed:
Remove PSSM Fade

akien-mga committed:
Merge pull request #79407 from zaevi/fix_web_websocket-close-reason

[Web] Fix WebSocket returning empty close-reason.

akien-mga committed:
Merge pull request #80182 from mikemar10/mikemar10/update_documentation_enetconnection_eventreceive

Fix packet details of ENetConnection EventType EVENT_RECEIVE documentation

akien-mga committed:
Merge pull request #82742 from kitbdev/empty-dock-layout

Fix loading floating dock layout

akien-mga committed:
Merge pull request #83038 from jsjtxietian/update-doc-mention-spinbox-reset-habeviour

Update SpinBox documentation to include resetting to min/max behaviour

akien-mga committed:
Merge pull request #83177 from timothyqiu/the-theme-identity

Don't auto translate theme type list

akien-mga committed:
Merge pull request #83189 from jsjtxietian/fix-shader-node-parameter-connect-to-vary-error

Fix parameter shader node not declared when only connected to a VaryingSetter

akien-mga committed:
Merge pull request #83194 from jsjtxietian/fix-bool-varying-code-gen

Fix bool varying's generated code will be modified with flat

akien-mga committed:
Merge pull request #83202 from Repiteo/vector4-projection-marshall-fix

Add missing double-precision flag for Vector4 & Projection in `encode_variant`

akien-mga committed:
Merge pull request #83210 from alfredbaudisch/manager-search-box-enter

Project Manager: Open project when "Enter" is pressed when the search box is focused

akien-mga committed:
Merge pull request #83211 from clayjohn/cleanup-mesh-stuff

Cleanup instances of using uint32_t for mesh formats

akien-mga committed:
Merge pull request #83216 from dalexeev/code-editor-fix-region-doc-comment

Code Editor: Fix regression with using doc comments for code regions

akien-mga committed:
Merge pull request #83233 from timothyqiu/babel

Disable port name auto translation in Visual Shader editor

akien-mga committed:
Merge pull request #83243 from bruvzg/macos_sys_def_menu_items

[macOS] Fix crash when using system default menu shortcuts.

akien-mga committed:
Merge pull request #83244 from KoBeWi/tile_map_is_now_tile_set_and_vice_versa

Swap TileMap and TileSet buttons

dalexeev committed:
GDScript: Fix unresolved datatype for incomplete expressions

akien-mga committed:
Merge pull request #80605 from bruvzg/bmp_font_scale

[Bitmap fonts] Add support for scaling.

akien-mga committed:
Merge pull request #82957 from bruvzg/fs_case_check

Add method to check if filesystem is case sensitive.

capnm committed:
ThorVG: Update to v0.11.1

https://github.com/thorvg/thorvg/releases/tag/v0.11.1

Godot related:
+ [SwEngine] Enhanced the quality of the dash line corners.
+ [SVG] Fixed a regression bug related to maskContentUnit,
	userSpaceOnUse/objectBoundingBox.

Fixes #82982 icons.
Fixes #83198 svg ← png/jpg.

KoBeWi committed:
Don't try updating wrong NodePaths in resources

Commit:
Fix unused variables warning when `touch` is disabled

bruvzg committed:
[Linux] Implement DirAccess.is_case_sensitive for EXT4 and F2FS.

Faless committed:
[MP] Fix "on change" indexed properties

Watchers were still using Object::get instead of Object::get_indexed.

groud committed:
Allow disabling the built-in tilemap navigation

DarioSamo committed:
Rewrite the GPU Lightmapper's indirect lighting logic to match Godot 3.5's CPU Lightmapper.

Port over the logic from Godot 3.5 for indirect lighting. This should fix many issues about indirect bounces causing more energy and improve the overall quality of the result.

bronsonholden committed:
Remove vertical scrollbar padding from line width calc

Visibility of the vertical scrollbar is already accounted for in
`_update_wrap_at_column` which in turn affects max line width of
the text area.

dsnopek committed:
GDExtension: Prevent issues with the editor trying to reload GDExtensions through its usual mechanism

bronsonholden committed:
Add test justifying change to expected horizontal scroll position

akien-mga committed:
Merge pull request #80916 from ajreckof/fix-deferred-mode-not-working-for-color-picker-sliders

Fix ColorPicker deferred mode not working for sliders.

akien-mga committed:
Merge pull request #82068 from DarioSamo/lightmapper-indirect-bounces

Rewrite the GPU Lightmapper's indirect logic to match Godot 3.5's CPU Lightmapper.

akien-mga committed:
Merge pull request #82381 from dervus/issue-81916

Fix grid snapping for box shape gizmos

akien-mga committed:
Merge pull request #83179 from clayjohn/tangents-sanitize

Sanitize tangents when creating mesh surfaces to avoid triggering the compressed mesh path in the shader

akien-mga committed:
Merge pull request #83239 from Daylily-Zeleen/Fixs-missing-editor-singletons-when-dump-extension-api

Fix missing editor singletons when dumping extension api

akien-mga committed:
Merge pull request #83249 from zaevi/fix_csharp_lookup_singleton_instance

C#: Fix lookup for singleton instance types.

akien-mga committed:
Merge pull request #83252 from mrjustaguy/opengl-fade-fix

Fix OpenGL Directional Shadow Last Split Fading

akien-mga committed:
Merge pull request #83264 from Faless/mp/timer_and_reset_fixes

[MP] Fix synchronizer init and reset

akien-mga committed:
Merge pull request #83265 from programneer/unused-vars-touch-fix

X11: Fix unused variables warning when `touch` is disabled

akien-mga committed:
Merge pull request #83266 from bruvzg/linux_case_check

[Linux] Implement DirAccess.is_case_sensitive for EXT4 and F2FS.

akien-mga committed:
Merge pull request #83273 from groud/disable_tilemap_navigation

Allow disabling the built-in tilemap navigation

akien-mga committed:
Merge pull request #83279 from Faless/mp/fix_delta_indexed

[MP] Fix "on change" indexed properties

akien-mga committed:
Merge pull request #80452 from strellydev/ogg-loop-offset-pop-fix

Fix OGG audio loop offset pop

Wierdox committed:
Fix phantom tab right button.

Rindbee committed:
Fix cannot update remote after disabling `use_global_coordinates` in `RemoteTransform2D`

Due to the optimization in `CanvasItem`, `global_transform` is only
updated when `get_global_transform()` is called, and then notify
`NOTIFICATION_TRANSFORM_CHANGED`. That is, in the case where
`global_transform` is not obtained, the notification will not be sent.

So we use `NOTIFICATION_LOCAL_TRANSFORM_CHANGED` in this case. Use in
combination to prevent certain optimizations.

Same change for `RemoteTransform3D`, to prevent the same optimization
from being used in `Node3D` in the future.

rburing committed:
Add brief description in GDExtension API dump with docs

Also rename the "documentation" key to "description" for internal
consistency.

AThousandShips committed:
Make error suggestion less ambiguous

KoBeWi committed:
Fix FindReplaceBar focus problems

Malcolmnixon committed:
Fix enabling NavigationRegion3D saved disabled

This PR fixes issue #83364 by always registering the region when it enters the tree, and instead using region_set_enabled to control whether its enabled. The same logical changes have been applied to NavigationRegion2D.

Lunarisnia committed:
Fix disabling depth prepass breaks opaque material

Lunarisnia committed:
Fix multiple comment delimiter break toggle comment shortcut

SaracenOne committed:
Disallow 'Make Local' command on inherited nodes.

bitsawer committed:
Fix Mobile renderer shader instance uniform access

YeldhamDev committed:
Disable "Edit Transitions..." item if no animations are present

touilleMan committed:
Fix comment in gdextension_interface.h

Rubonnek committed:
Show AnimationMixer warning for non-numeric types only when relevant

raulsntos committed:
C#: Allow exporting games without C#

When exporting a game that contains a C# solution, a feature is added so the exported game can check if it should initialize the .NET module. Otherwise, the module initialization is skipped so games without C# won't check for the assemblies and won't show alerts when they're missing.

kleonc committed:
Fix screen center position returned for rotated Camera2D

akien-mga committed:
Merge pull request #76244 from kgrubbs22/variant-test-updates

Added Unit Tests for Variant for Operator Overloading

akien-mga committed:
Merge pull request #82627 from kleonc/animation-editor-audio-clip-offset-sliders-limit-to-length

Limit animation audio clip inspector offset sliders to clip length

akien-mga committed:
Merge pull request #82846 from SaracenOne/network_authority_instance_placeholders

Copy network authority when instancing placeholders

akien-mga committed:
Merge pull request #82868 from SaracenOne/multi_session_debugger_fixes

Fix debugger behaviour with multi-session debugging

akien-mga committed:
Merge pull request #82937 from Calinou/editor-primitivemesh-unwrap-uv2

Enable UV2 on primitive meshes when using the MeshInstance3D context menu

akien-mga committed:
Merge pull request #83146 from pancelor/fix-tilemap-liveupdate

Fix tilemap live editing while game is running

akien-mga committed:
Merge pull request #83173 from m4gr3d/fix_gestures_properties_retrieval_timing

Fix the timeframe when the Android gestures properties are retrieved

akien-mga committed:
Merge pull request #83263 from KoBeWi/invalid_node﹖oh_no…_anyway

Don't try updating wrong NodePaths in resources

akien-mga committed:
Merge pull request #83296 from Wierdox/fix_phantom_tab_right_button

Fix phantom tab right button

akien-mga committed:
Merge pull request #83318 from rburing/gdextension_dump_brief_description

Add brief description in GDExtension API dump with docs

akien-mga committed:
Merge pull request #83327 from AThousandShips/doc_suggestion

Make error suggestion less ambiguous

akien-mga committed:
Merge pull request #83371 from Lunarisnia/fix/disabling-depth-prepass

Fix disabling depth prepass break opaque materials

akien-mga committed:
Merge pull request #83400 from bitsawer/fix_mobile_instance_uniforms

Fix Mobile renderer shader instance uniform access

akien-mga committed:
Merge pull request #83402 from YeldhamDev/edit_transitions_hide

Disable "Edit Transitions..." item if no animations are present

akien-mga committed:
Merge pull request #83417 from Rubonnek/show-animation-mixer-warning-when-relevant

Show AnimationMixer warning for non-numeric types only when relevant

akien-mga committed:
Merge pull request #83422 from raulsntos/dotnet/export-projects-without-csharp

C#: Allow exporting games without C#

mhilbrunner committed:
Docs: Fix link to Android Gradle build tutorial

akien-mga committed:
Merge pull request #73502 from Kurble/text-edit-scrolling-precision

Fix code editor scrolling experience on track pads

akien-mga committed:
Merge pull request #82056 from ErezShahaf/Axisjustpressedjitterfix

Fix `Input.is_action_just_pressed` flicker on joypad axes

akien-mga committed:
Merge pull request #83137 from jsjtxietian/close-shader-file-in-visual-editor-when-delete-file

Close shader in Shader Editor tab when deleting shader file in FileSystem panel

akien-mga committed:
Merge pull request #83323 from Rindbee/fix-not-update-remote-local-transform-in-2d

Fix cannot update remote after disabling `use_global_coordinates` in `RemoteTransform2D`

akien-mga committed:
Merge pull request #83365 from Malcolmnixon/fix-disabled-nav-region

Fix enabling NavigationRegion3D saved disabled

akien-mga committed:
Merge pull request #83386 from SaracenOne/make_local_inherited

Disallow 'Make Local' command on inherited nodes.

akien-mga committed:
Merge pull request #83427 from kleonc/camera2d-screen-center-position-when-rotated

Fix screen center position returned for rotated Camera2D

akien-mga committed:
Merge pull request #83433 from mhilbrunner/fix-android-gradle-docs

Docs: Fix link to Android Gradle build tutorial

raulsntos committed:
Remove AnimationMixer bindings only bound in the editor

akien-mga committed:
Merge pull request #83281 from capnm/thorvg_0.11.1

ThorVG: update to v0.11.1

akien-mga committed:
Merge pull request #83335 from KoBeWi/focusing_search_power

Fix FindReplaceBar focus problems

akien-mga committed:
Merge pull request #83382 from Lunarisnia/fix/diff-comment-delimiter

Fix multiple comment delimiter break toggle comment shortcut

raulsntos committed:
Set icon_max_width in the ConnectionsDock tree

akien-mga committed:
Merge pull request #78882 from dalexeev/gds-fix-utility-func-err-msg

GDScript: Fix incorrect error message for utility functions

akien-mga committed:
Merge pull request #83123 from astillich/82998-propagate-base-class-exports

Fix modifying base script exports not propagating to derived scripts

akien-mga committed:
Merge pull request #83181 from bruvzg/cb_dlg

Add error messages to the native menu and file dialogs callback.

akien-mga committed:
Merge pull request #83257 from dalexeev/gds-fix-unresolved-type-for-incomplete-expressions

GDScript: Fix unresolved datatype for incomplete expressions

akien-mga committed:
Merge pull request #83285 from dsnopek/gdextension-double-reload

GDExtension: Prevent issues with the editor trying to reload GDExtensions through its usual mechanism

akien-mga committed:
Merge pull request #83440 from raulsntos/animation/remove-tool-bindings

Remove AnimationMixer bindings only bound in the editor

akien-mga committed:
Merge pull request #83447 from raulsntos/editor-iconography-2

Set `icon_max_width` in the ConnectionsDock tree

akien-mga committed:
Merge pull request #82797 from RandomShaper/fix_rbmap

Fix `RBMap`'s, iterator-based, `remove()`

Lunarisnia committed:
Fix grammar typo in GDScript error message

kleonc committed:
Clarify docs for operators performing `xform_inv`

touilleMan committed:
Merge pull request #83415 from touilleMan/fix-comment-gdextension_interface

Fix comment in `gdextension_interface.h`

KoBeWi committed:
Fix close button in FindReplaceBar

Eoin-ONeill-Yokai committed:
Fix bug where TileMap will not update material correct on assignment

This seemed to be a simple case of a missing condition in an if
statement with a few different flags. I'm not super familiar with the
tilemap code but hopefully this is the correct place to add this check.

Fixes #83474

QbieShay committed:
Fixed division by zero in particle process, initial ramp, and amount ratio+scale

groud committed:
Allow normal maps on TileMaps that use texture padding

AThousandShips committed:
Fix shortcut input for `EditorSceneTabs`

akien-mga committed:
Merge pull request #81196 from georgwacker/add-audiosession-options

Add project settings for AVAudioSessionCategory on iOS

akien-mga committed:
Merge pull request #82582 from akien-mga/openxr-1.0.30

openxr: Sync with upstream 1.0.30

akien-mga committed:
Merge pull request #83286 from bronsonholden/fix-text-edit-hscroll-jitter

Remove vertical scrollbar padding from line width calc

akien-mga committed:
Merge pull request #83455 from Lunarisnia/fix/gdscript-error-typo

Fix grammar typo in GDScript error message

akien-mga committed:
Merge pull request #83459 from KoBeWi/bug=false

Fix close button in FindReplaceBar

akien-mga committed:
Merge pull request #83461 from kleonc/docs-multiplication-operators-doing-xform_inv

Clarify docs for operators performing `xform_inv`

akien-mga committed:
Merge pull request #83475 from Eoin-ONeill-Yokai/bug/tilemap-materials

Fix bug where TileMap wouldn't update material correctly on assignment

akien-mga committed:
Merge pull request #83501 from AThousandShips/tabs_shortcut

Fix shortcut input for `EditorSceneTabs`

ElMikosch committed:
Fix unit suffix for HingeJoint3D's target velocity

kleonc committed:
Clarify C# docs for operators performing xform_inv

Commit:
Bump the java version to version 17

YuriSizov committed:
Fix incorrect offset of `PopupMenu` separator icons

akien-mga committed:
Merge pull request #80489 from raulsntos/dotnet/diagnostics

C#: Improve diagnostic messages and add help link

akien-mga committed:
Merge pull request #80523 from ElMikosch/fix_hingejoint3d_target_velocity_suffix

Fix unit suffix for `HingeJoint3D`'s target velocity

akien-mga committed:
Merge pull request #81774 from MewPurPur/improve-visible-on-screen-class-docs

Improve VisibleOnScreen classes' docs

akien-mga committed:
Merge pull request #82409 from KoBeWi/3_in_1

Document, cleanup and fix some theme properties

akien-mga committed:
Merge pull request #82858 from KoBeWi/read_between_the_comments

Provide translation strings for folder colors

akien-mga committed:
Merge pull request #83251 from Rindbee/fix-current_edited_scene-wrong-move

Fix saving wrong edited scene state when switching scene tabs

akien-mga committed:
Merge pull request #83514 from kleonc/docs-multiplication-operators-doing-xform_inv-csharp

Clarify C# docs for operators performing `xform_inv`

raulsntos committed:
Fix C# docs to use the proper XML

Replaces some BBCode that was accidentally left in the C# documentation
with the proper XML syntax.

raulsntos committed:
C#: Fix generated nested class order

SaracenOne committed:
Automatic reconnection of nodes in blend tree.

akien-mga committed:
Merge pull request #81638 from jsjtxietian/fix-spinBox-reset-text-when-redraw

Fix SpinBox will reset unsubmited text when redrawing

akien-mga committed:
Merge pull request #81643 from Rindbee/defer-rebuild-the-animations-tree-in-SpriteFramesEditor

Defer updating the animations Tree in SpriteFramesEditor to avoid crashes

akien-mga committed:
Merge pull request #83489 from groud/implement_tile_map_normals

Allow normal maps on TileMaps that use texture padding

akien-mga committed:
Merge pull request #83515 from m4gr3d/bump_java_version_main

Bump the java version to version 17

akien-mga committed:
Merge pull request #83517 from YuriSizov/for-NON-separator-items-add-some-padding-for-the-content

Fix incorrect offset of `PopupMenu` separator icons

akien-mga committed:
Merge pull request #83529 from raulsntos/dotnet/docs-xml

Fix C# docs to use the proper XML

akien-mga committed:
Merge pull request #83532 from raulsntos/dotnet/nested-class-generation

C#: Fix generated nested class order

akien-mga committed:
Merge pull request #83534 from SaracenOne/blend_tree_auto_disconnect

Automatic reconnection of nodes in blend tree.

akien-mga committed:
Merge pull request #82537 from KoBeWi/the_files_are_back_on_the_menu

Don't remove favorite files in EditorFileDialog

YuriSizov committed:
Fix StringName leaks in GDExtension, core, and editor themes

RandomShaper committed:
Bump version of Vulkan RD binary shader format

mrTag committed:
Removed navigation_mesh instantiation from constructor of NavigationPolygon

smix8 committed:
Fix NavRegion sync error messages

Fixes NavRegion sync error messages

akien-mga committed:
Merge pull request #83488 from QbieShay/qbe/fix-particle-pause

Fixed multiple particle issues: division by zero, color ramp override, scale dependent on amount ratio

akien-mga committed:
Merge pull request #83562 from YuriSizov/core-our-vessel-is-not-seaworthy

Fix StringName leaks in GDExtension, core, and editor themes

akien-mga committed:
Merge pull request #83563 from RandomShaper/vk_shader_hist

Bump version of Vulkan RD binary shader format

akien-mga committed:
Merge pull request #83574 from smix8/navregion_err_msg_4.x

Fix NavRegion sync error messages

akien-mga committed:
Merge pull request #83568 from mrTag/navigationregion2d_fix

Fixing "Navigation map synchronization error" when using NavigationRegion2D

RPicster committed:
Merge branch 'triplanar-normal-maps-model-matrix' of https://github.com/RPicster/godot into triplanar-normal-maps-model-matrix
